Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionCocoa.mm

```diff
@@ -1094,6 +1114,66 @@ static _WKWebExtensionError toAPI(WebExtension::Error error)
 #endif
 }
 
+bool WebExtension::hasOptionsPage()
+{
+    populatePagePropertiesIfNeeded();
+    return !!m_optionsPagePath;
+}
+
+bool WebExtension::hasOverrideNewTabPage()
+{
+    populatePagePropertiesIfNeeded();
+    return !!m_overrideNewTabPagePath;
+}
+
+NSString *WebExtension::optionsPagePath()
+{
+    populatePagePropertiesIfNeeded();
+    return m_optionsPagePath.get();
+}
+
+NSString *WebExtension::overrideNewTabPagePath()
+{
+    populatePagePropertiesIfNeeded();
+    return m_overrideNewTabPagePath.get();
+}
+
+void WebExtension::populatePagePropertiesIfNeeded()
+{
+    if (!manifestParsedSuccessfully())
+        return;
+
+    if (m_parsedManifestPageProperties)
```

Comment:
Is this thread safe? Or do we know that this will only ever be hit on the main thread? I'm wondering if we need to guard against this being called from a work queue.

Reply:
This is a WebKit APIObject, so it is expected to be main-thread only. We don't use threads here.